app/presenters/card_orders_export_presenter.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+class CardOrdersExportPresenter < WasteCarriersEngine::BasePresenter
+
+  def initialize(model)
+    @order_item_log = model
+    @registration = WasteCarriersEngine::Registration.find(model.registration_id)
+
+    # Address fields need to be presented as a set but may be queried individually,
+    # so prepare them once for lookup on initialiation
+    @registered_address = present_address(
+      @registration.addresses.select { |a| a.addressType == "REGISTERED" }[0],
+      "registered"
+    )
+    @contact_address = present_address(
+      @registration.addresses.select { |a| a.addressType == "POSTAL" }[0],
+      "contact"
+    )
+
+    super(model)
+  end
+
+  def reg_identifier
+    @order_item_log.registration_id
+  end
+
+  def date_of_issue
+    @order_item_log.activated_at
+  end
+
+  def carrier_name
+    case @registration.tier
+    when "LOWER"
+      @registration.company_name
+    when "UPPER"
+      case @registration.business_type
+      when "soleTrader"
+        "#{@registration.key_people[0].first_name} #{@registration.key_people[0].last_name}"
+      else
+        @registration.company_name
+      end
+    else
+      ""
+    end
+  end
+
+  def company_name
+    @registration.company_name
+  end
+
+  def registration_type
+    @registration.registration_type
+  end
+
+  def registration_date
+    @registration.metaData.dateRegistered
+  end
+
+  def expires_on
+    @registration.expires_on
+  end
+
+  def contact_phone_number
+    @registration.phone_number
+  end
+
+  # Define address-level fields dynamically
+  address_keys = %w[line_1 line_2 line_3 line_4 line_5 line_6 town_city postcode country].freeze
```

Comment:
I'm probably not understanding properly - do we use `line_6` in the code?

Reply:
The required columns as per https://eaflood.atlassian.net/browse/RUBY-1066?focusedCommentId=372548 include a line_6 column. In practice this will be blank. I'll check whether this is in fact required by the card printer.